homeassistant/components/cloud/alexa_config.py

```diff
@@ -198,35 +200,41 @@ def _migrate_alexa_entity_settings_v1(self):
             # Don't migrate if there's a YAML config
             return
 
-        entity_registry = er.async_get(self.hass)
-
-        for entity_id, entry in entity_registry.entities.items():
-            if CLOUD_ALEXA in entry.options:
-                continue
-            options = {"should_expose": self._should_expose_legacy(entity_id)}
-            entity_registry.async_update_entity_options(entity_id, CLOUD_ALEXA, options)
+        for state in self.hass.states.async_all():
```

Comment:
Should we also migrate any entity that is stored inside the cloud preferences?

Reply:
Since we don't use state, you could just iterate over entity IDs btw.

Configured entities available as
`for entity_id in self._prefs.alexa_entity_configs`